packages/ingest/granule.js

```diff
@@ -106,20 +108,34 @@ class Discover {
    * @private
    */
   fileTypeConfigForFile(file) {
-    return this.collection.files.find((fileTypeConfig) => file.name.match(fileTypeConfig.regex));
+    return this.collection.files.find((fileTypeConfig) =>
+      file.name.match(fileTypeConfig.regex));
   }
 
   /**
-   * Discover new granules
+   * Returns a possibly empty array of discovered granules.  Each granule will
+   * contain a possibly empty array of files, influenced by the `boolean`
```

Comment:
We could, but that's what led to the start of this whole thing to begin with. The original issue was created because the user (apparently left the team, so that's why we now are speaking with Abdelhak) was seeing "Discovered 0 granules" when they knew they should have discovered granules. They were getting 0 granules because of the whole problem with files not matching any regexes at discovery (because they files are renamed later, and then match a regex at that time). To avoid confusion in the future, I suggest we still indicate a discovered granule so it's still visible in the dashboard, but then they can see that there are (possibly) no related files and investigate from there. Either way, they will need to "debug" the "0 files issue", but at least they will see a granule with no files rather than no granule at all.

Reply:
Ok, I agree, "0 file granule" can indicate a configuration issue.